dist/common/scripts/scylla_util.py

```diff
@@ -117,6 +117,239 @@ def curl(url, headers=None, byte=False, timeout=3, max_retries=5, retry_interval
+        return self.__memoryGB
+
+    def instance_size(self):
+        """Returns the size of the instance we are running in. i.e.: 2"""
+        instancetypesplit = self.instancetype.split("_")
+        return instancetypesplit[2] if len(instancetypesplit)>2 else 0
+
+    def instance_class(self):
+        """Returns the class of the instance we are running in. i.e.: n2"""
+        return self.instancetype.split("_")[0]
+
+    def instance_purpose(self):
+        """Returns the purpose of the instance we are running in. i.e.: standard"""
+        return self.instancetype.split("_")[1]
+
+    m1supported="m1-megamem-96"
+
+    def is_unsupported_instance_class(self):
+        """Returns if this instance type belongs to unsupported ones for nvmes"""
+        if self.instancetype == self.m1supported:
+            return False
+        if self.instance_class() in ['e2', 'f1', 'g1', 'm2', 'm1']:
+            return True
+        return False
+
+    def is_supported_instance_class(self):
+        """Returns if this instance type belongs to supported ones for nvmes"""
+        if self.instancetype == self.m1supported:
+            return True
+        if self.instance_class() in ['n1', 'n2', 'n2d' ,'c2']:
+            return True
+        return False
+
+
+    def is_recommended_instance_size(self):
+        """if this instance has at least 2 cpus, it has a recommended size"""
+        if int(self.instance_size()) > 1:
+            return True
+        return False
+
+    @staticmethod
+    def get_file_size_by_seek(filename):
+        "Get the file size by seeking at end"
+        fd= os.open(filename, os.O_RDONLY)
+        try:
+            return os.lseek(fd, 0, os.SEEK_END)
+        finally:
+            os.close(fd)
+
+    # note that GCP has 3TB physical devices actually, which they break into smaller 375GB disks and share the same mem with multiple machines
+    # this is a reference value, disk size shouldn't be lower than that
+    AZURE_NVME_DISK_SIZE_2020=375
+
+    @property
+    def firstNvmeSize(self):
+        """return the size of first non root NVME disk in GB"""
+        if self.__firstNvmeSize is None:
+            ephemeral_disks = self.getEphemeralOsDisks()
+            firstDisk = ephemeral_disks[0]
+            firstDiskSize = self.get_file_size_by_seek(os.path.join("/dev/", firstDisk))
+            firstDiskSizeGB = firstDiskSize/1024/1024/1024
+            if firstDiskSizeGB >= self.AZURE_NVME_DISK_SIZE_2020:
+                self.__firstNvmeSize = firstDiskSizeGB
+            else:
+                raise Exception("First nvme is smaller than lowest expected size. ".format(firstDisk))
+        return self.__firstNvmeSize
+
+    def is_recommended_instance(self):
+        if not self.is_unsupported_instance_class() and self.is_supported_instance_class() and self.is_recommended_instance_size():
+            # at least 1:2GB cpu:ram ratio , GCP is at 1:4, so this should be fine
```

Comment:
Lets either (a) figure out what CPU:RAM ratio we want for Azure (you can talk to @avikivity, for example), or (b) just drop the checks for now.